changes/2024-TBD_cache-across-hierarchy-keyrings/change.md

```diff
@@ -0,0 +1,243 @@
+the CMC MAY cache Branch Keys protected by different
+KMS Relationships.
+
+This simplifies Multiple KMS Relationship MPL Consumers,
+as they do not need to stand up LRU Cache of Hierarchy Keyrings.
+
+Instead, they may maintain one CMC.
+They still create a Hierarchy Keyring instance per KMS Relationship,
+and they MUST use the correct Keyring to retrieve material
+from the Cache.
+
+But they need not maintain many Keyrings;
+only the common cache.
+
+Cache misses will populate the cache via
+the Hierarchy Keyring that requested the material.
+
+However, Cache Entries MUST only be served IF
+the request is being served under the same KMS Relationship,
+which is the most secure condition,
+as it prevents a Confused Deputy scenario
+(detailed in **Confused Deputy 1**).
+
+To facilitate this,
+the Cache Entry Identifier MUST
+describe the KMS Relationship,
+as well as the Logical Key Store Name,
+Branch Key ID,
+and Branch Key Name.
+
+The KeyStore constructor takes an optional [ID](../../framework/branch-key-store.md#keystore-id)
+parameter at construction.
+
+We can use this [ID](../../framework/branch-key-store.md#keystore-id)
+to label the KMS Relationship of a particular KeyStore instance.
+
+MPL Consumers who want to garbage collect KeyStore instances,
+but still retain the ability to serve cached results,
+MAY then set this [ID](../../framework/branch-key-store.md#keystore-id)
+intelligently.
+
+For example, imagine we have `N` KMS Relationships,
+one of which is `X`,
+which are used when retrieving Branch Keys, `n'`,
+one of which is `x'`.
+
+A common Cryptographic Materials Cache is established
+across all `N` Hierarchical Keyrings used to serve `n'`.
+
+But the Keyrings are routinely garbage collected,
+only the common cache remains "in scope" indefinitely.
+
+Whenever `x'` is needed,
+a new KMS Client is created that respects relationship `X`,
+which is then used to create KeyStore Identified as `X`,
+and the KeyStore is used to create a Hierarchical Keyring
+that uses the common Cache.
+
+All entries in this common Cache from that hierarchal Keyring
+MUST have a cache identifier that uniquely represents `X`.
+
+When the Hierarchical Keyring for `X` gets garbage collected,
+the cache entries MAY remain.
+
+But to retrieve an entry,
+the MPL Consumer MUST recreate
+
+> a KMS Client is created that respects relationship `X`,
+> which is then used to create KeyStore Identified as `X`,
+> and the KeyStore is used to create a Hierarchical Keyring
+> that uses the common Cache.
+
+If the Cache Entry is still valid
+(the TTL has not expired),
+it can be served.
+
+If not, the recreated Hierarchy Keyring
+is used to refresh it.
+
+To push MPL Consumers in the correct direction,
+AWS Crypto Tools could write a helper method,
+that creates KeyStores with KMS Clients
+of a particular KMS Relationship,
+and sets the KeyStore's ID deterministically
+on on the properties of the KMS Relationship.
+
+## Out of Scope
+
+TODO
+
+## Motivation
+
+TODO
```

Comment:
I think I am going to break our convention, and move Motivation to be right behind Summary.